### PR TITLE
Various sorting benchmarks

### PR DESCRIPTION
**Pull request by AndyAyersMS**
TreeSort, HeapSort, QuickSort, BubbleSort, BubbleSort2

### CODE REVIEW

**Comment by AndyAyersMS** [comment]
@cmckinsey PTAL

File: tests/src/JIT/Performance/CodeQuality/BenchI/QuickSort/QuickSort.cs [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+        int[] buffer = new int[MAXNUM];
+        for (int j = 0; j < MAXNUM; ++j) {
+            int temp = Random(MODULUS);
+            if (temp < 0){
+                temp = (-temp);
+            }
+            buffer[j] = temp;
+        }
+        Quick(0, MAXNUM - 1, buffer);
+        return true;
```
Comment (cmckinsey) [inline tests/src/JIT/Performance/CodeQuality/BenchI/QuickSort/QuickSort.cs]: Would be straightforward to add checking code that the array was sorted like the other sort benchmarks.

**Comment by cmckinsey** [comment]
LGTM after that.

**Comment by AndyAyersMS** [comment]
Added validation to QuickSort.

**Comment by cmckinsey** [comment]
LGTM.

**Comment by AndyAyersMS** [comment]
OSX B&T CI seems broken, am going to ignore the failure.

**Pull request merged**